packages/build/tests/blobs_upload/tests.js

```diff
@@ -0,0 +1,99 @@
+import { version as nodeVersion } from 'node:process'
+
+import { BlobsServer, getDeployStore } from '@netlify/blobs'
+import { Fixture, normalizeOutput } from '@netlify/testing'
+import test from 'ava'
+import getPort from 'get-port'
+import semver from 'semver'
+import tmp from 'tmp-promise'
+
+const TOKEN = 'test'
+
+test.beforeEach(async (t) => {
+  const port = await getPort()
+  t.context.blobRequestCount = { set: 0, get: 0 }
+
+  const tmpDir = await tmp.dir()
+  t.context.blobServer = new BlobsServer({
+    port,
+    token: TOKEN,
+    directory: tmpDir.path,
+    onRequest: ({ type }) => {
+      t.context.blobRequestCount[type] = (t.context.blobRequestCount[type] || 0) + 1
+    },
+  })
+
+  await t.context.blobServer.start()
+
+  process.env.NETLIFY_BLOBS_CONTEXT = Buffer.from(
+    JSON.stringify({
+      edgeURL: `http://localhost:${port}`,
+    }),
+  ).toString('base64')
+})
+
+test.afterEach.always(async (t) => {
+  await t.context.blobServer.stop()
+  delete process.env.NETLIFY_BLOBS_CONTEXT
+})
+
+test("blobs upload, don't run when deploy id is provided and no files in directory", async (t) => {
+  const output = await new Fixture('./fixtures/src_empty')
+    // Passing `offline: true` to avoid fetching the configuration from the API
+    .withFlags({ deployId: 'abc123', token: TOKEN, offline: true })
+    .runWithBuild()
+
+  t.is(t.context.blobRequestCount.set, 0)
+
+  t.snapshot(normalizeOutput(output))
```

Comment:
Wondering if we still need a full snapshot test or if we could do some partial assertion on the output and that would be enough? (check there's nothing referencing blob upload or something). Not a blocker.

Reply:
I think we can try to assert on presence or lack of presence of messages. I'll give it a go